neat_ml/tests/test_preprocessing.py

```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+import cv2
+import matplotlib
+import pytest
+from pytest import MonkeyPatch
+from typing import Any
+import pooch  # type: ignore[import-untyped]
+import os
+import logging
+
+from matplotlib.testing.compare import compare_images
+
+from neat_ml.opencv import preprocessing as pp
+
+matplotlib.use("Agg")
+
+def test_process_directory_single_image(
+    tmp_path: Path,
+    reference_images : list,
+):
+    raw_input = reference_images[3] 
+    pp.process_directory(pooch.os_cache("test_images"), tmp_path)
+    processed_tiff = tmp_path / os.path.basename(raw_input)
+    img = cv2.imread(str(processed_tiff), cv2.IMREAD_GRAYSCALE)
+    actual_png = tmp_path / "raw_processed.png"
+    assert img is not None
```

Comment:
probably don't need this `assert`, the genuine image comparison below should suffice